### PR TITLE
Update all editor viewports after settings changes

### LINKED ISSUE #18430: Spatial editor's view settings aren't applied immediately after changing them
**Godot version:** Git https://github.com/godotengine/godot/commit/e8ff602381edc2bf8ba56de85b50c275c17f9a0c

**OS/device including version:** Fedora 27

**Issue description:** The spatial editor's view settings aren't applied immediately after changing them, or even after clicking the Close button of the Settings dialog. Changes will be displayed only once the editor camera is moved by the user somehow, even if the editor update mode is set to **Update Always**.

**Steps to reproduce:** Switch to the editor's 3D view, go to the **View** → **Settings** menu at the top of the 3D viewport, adjust FOV/Z-Near/Z-Far values then close the settings menu. The displayed values won't change until the editor camera is moved somehow (for instance, by looking around or zooming).

### LINKED ISSUE #34435: Changing view settings in 3D, should update all viewports
**Godot version:**
Godot 3.2 Beta 3
**OS/device including version:**
Ubuntu 19.10
**Issue description:**
Godot should update viewports immediately when user change e.g. Z view setting

This patch works, but isn't the best solution
```
diff --git a/editor/plugins/spatial_editor_plugin.cpp b/editor/plugins/spatial_editor_plugin.cpp
index 10e4559805..2da9351e05 100644
--- a/editor/plugins/spatial_editor_plugin.cpp
+++ b/editor/plugins/spatial_editor_plugin.cpp
@@ -4000,6 +4000,7 @@ SpatialEditorViewportContainer::View SpatialEditorViewportContainer::get_view()
 void SpatialEditorViewportContainer::_bind_methods() {
 
        ClassDB::bind_method("_gui_input", &SpatialEditorViewportContainer::_gui_input);
+
 }
 
 SpatialEditorViewportContainer::SpatialEditorViewportContainer() {
@@ -5514,6 +5515,13 @@ void SpatialEditor::_node_removed(Node *p_node) {
                selected = NULL;
 }
 
+void SpatialEditor::_update_viewports(){
+       for(unsigned int i=0;i<VIEWPORTS_COUNT;i++){
+               viewports[i]->hide();
+               viewports[i]->show();
+    }
+}
+
 void SpatialEditor::_register_all_gizmos() {
        add_gizmo_plugin(Ref<CameraSpatialGizmoPlugin>(memnew(CameraSpatialGizmoPlugin)));
        add_gizmo_plugin(Ref<LightSpatialGizmoPlugin>(memnew(LightSpatialGizmoPlugin)));
@@ -5553,6 +5561,7 @@ void SpatialEditor::_bind_methods() {
        ClassDB::bind_method("_refresh_menu_icons", &SpatialEditor::_refresh_menu_icons);
        ClassDB::bind_method("_update_camera_override_button", &SpatialEditor::_update_camera_override_button);
        ClassDB::bind_method("_update_camera_override_viewport", &SpatialEditor::_update_camera_override_viewport);
+    ClassDB::bind_method("_update_viewports", &SpatialEditor::_update_viewports);
 
        ADD_SIGNAL(MethodInfo("transform_key_request"));
        ADD_SIGNAL(MethodInfo("item_lock_status_changed"));
@@ -5860,6 +5869,8 @@ SpatialEditor::SpatialEditor(EditorNode *p_editor) {
        settings_zfar->set_value(EDITOR_DEF("editors/3d/default_z_far", 1500));
        settings_vbc->add_margin_child(TTR("View Z-Far:"), settings_zfar);
 
+    settings_dialog->connect("confirmed",this,"_update_viewports");
+
        /* XFORM DIALOG */
 
        xform_dialog = memnew(ConfirmationDialog);
diff --git a/editor/plugins/spatial_editor_plugin.h b/editor/plugins/spatial_editor_plugin.h
index 65e3c32ca8..599d2146eb 100644
--- a/editor/plugins/spatial_editor_plugin.h
+++ b/editor/plugins/spatial_editor_plugin.h
@@ -651,6 +651,8 @@ private:
 
        void _register_all_gizmos();
 
+    void _update_viewports();
+
        SpatialEditor();
 
        bool is_any_freelook_active() const;
```
**Steps to reproduce:**
Now:
jfile
After patch:
jfile

### PR DESCRIPTION
**Pull request by JFonS**
Fixes #34435.
Fixes #18430.

### CODE REVIEW

File: editor/plugins/spatial_editor_plugin.cpp [hunk 1]
```diff
@@ -6241,6 +6241,10 @@ SpatialEditor::SpatialEditor(EditorNode *p_editor) {
 	settings_zfar->set_value(EDITOR_DEF("editors/3d/default_z_far", 1500));
 	settings_vbc->add_margin_child(TTR("View Z-Far:"), settings_zfar);
 
+	for (int i = 0; i < VIEWPORTS_COUNT; ++i) {
+        	settings_dialog->connect("confirmed", callable_mp(viewports[i], &SpatialEditorViewport::_update_camera), varray(0.0));
```
Comment (YeldhamDev) [inline editor/plugins/spatial_editor_plugin.cpp]: Space indentation here, should be fully tabbed.

**Comment by JFonS** [comment]
Fixed the CI issues.

**Comment by Calinou** [comment]
Does this also close #18430 (i.e. when using a single viewport)?

**Comment by JFonS** [comment]
@Calinou It does fix it :)

**Comment by akien-mga** [comment]
Thanks!

**Comment by akien-mga** [comment]
Cherry-picked for 3.2.2.

**Pull request merged**